cpp/cmake/CPM.cmake

```diff
@@ -0,0 +1,19 @@
+set(CPM_DOWNLOAD_VERSION c3c7e2d9a3b890511891ffc415d7ea23c689068a) # 0.27.3
```

Comment:
We should actually bump to 0.27.4 as there was some nice fixes related to specifying a source directory via cmake variables and being able to pass options through.